ion/integration_test.go

```diff
@@ -0,0 +1,808 @@
+	// Nested for loops to evaluate each ionItem value with all the other values in the list/struct/sexp
+	for i := 0; i < len(values); i++ {
+		for j := i + 1; j < len(values); j++ {
+			if i == j {
+				continue
+			}
+			if eq {
+				if !reflect.DeepEqual(values[i].value, values[j].value) ||
+					!reflect.DeepEqual(values[i].annotations, values[j].annotations) ||
+					!reflect.DeepEqual(values[i].ionType, values[j].ionType) {
+					t.Errorf("Equivalency test failed. All values should be interpreted as"+
+						"equal for %v and %v", values[i].value, values[j].value)
+				}
+			} else {
+				if reflect.DeepEqual(values[i].value, values[j].value) &&
+					reflect.DeepEqual(values[i].annotations, values[j].annotations) &&
+					reflect.DeepEqual(values[i].ionType, values[j].ionType) {
+					t.Errorf("Non-Equivalency test failed. Values should not be interpreted as"+
+						"equal for %v and %v", values[i].value, values[j].value)
+				}
+			}
+		}
+	}
+}
+
+// Read and load the files in testing path and pass them to testing functions.
+func readFilesAndTest(t *testing.T, path string, skipList []string, testingFunc testingFunc) {
+	files, err := ioutil.ReadDir(path)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for _, file := range files {
+		fp := filepath.Join(path, file.Name())
+		if file.IsDir() {
+			readFilesAndTest(t, fp, skipList, testingFunc)
+		} else if skipFile(skipList, file.Name()) {
+			continue
+		} else {
+			t.Run(fp, func(t *testing.T) {
+				testingFunc(t, fp)
+			})
+		}
+	}
+}
+
+func loadFile(t *testing.T, path string) []byte {
+	b, err := ioutil.ReadFile(path)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return b
+}
+
+// Files with extensions other than "ion" or "10n", or in skip list. Return True to skip the file, false otherwise.
+func skipFile(skipList []string, fn string) bool {
+	ionFile := strings.HasSuffix(fn, "ion") ||
+		strings.HasSuffix(fn, "10n")
+
+	return !ionFile || isInSkipList(skipList, fn)
+}
+
+func isInSkipList(skipList []string, fn string) bool {
+	for _, a := range skipList {
+		if a == fn {
+			return true
+		}
+	}
+	return false
+}
+
+// Read all the values in the reader and write them in the writer
+func writeToWriterFromReader(t *testing.T, reader Reader, writer Writer) {
+	for reader.Next() {
+		name := reader.FieldName()
+		if name != "" {
+			writer.FieldName(name)
+		}
+
+		an := reader.Annotations()
+		if len(an) > 0 {
+			writer.Annotations(an...)
+		}
+
+		switch reader.Type() {
+		case NullType:
+			err := writer.WriteNull()
+			if err != nil {
+				t.Errorf("Something went wrong when writing Null value. " + err.Error())
+			}
+
+		case BoolType:
+			val, err := reader.BoolValue()
+			if err != nil {
+				t.Errorf("Something went wrong when reading Boolean value. " + err.Error())
+			}
+			err = writer.WriteBool(val)
+			if err != nil {
+				t.Errorf("Something went wrong when writing Boolean value. " + err.Error())
+			}
+
+		case IntType:
+			val, err := reader.Int64Value()
```

Comment:
Caught this while working on int support, I believe this should be BigIntValue() and then we also write bigint.

Reply:
Changed. Thanks.